### PR TITLE
Add pytriangle, python wrapper to the Triangle C-API

### PR DESCRIPTION
**Pull request by rbeucher**
Checklist
- [X] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml"
- [X] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example)
- [X] Source is from official source
- [X] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged)
- [X] If static libraries are linked in, the license of the static library is packaged.
- [X] Build number is 0
- [X] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details)
- [X] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there

Please consider adding this package to conda-forge. There is a "triangle" package but this refers to the C-API. That package contains the python wrappers.

### CODE REVIEW

**Comment by rbeucher** [comment]
@conda-forge/bioconda-recipes I need this package before submitting a new package that depends on it.
I am not sure how I can use dependencies from bioconda in a conda-forge staged-recipe...

Thanks for your help

**Comment by mbargull** [comment]
Hi @rbeucher, IIUC there is
1. `conda-forge::triangle`: the underlying library,
2. `bioconda::pytriangle`: which is https://pypi.org/project/pytriangle/ , apparently another Python wrapper for Triangle
   (and this one should also live in `conda-forge` rather than `bioconda` -- but that's something for another day and nothing you have to handle),
3. https://pypi.org/project/triangle/ , the wrapper you want to add.

Given all of that, I'd go with `python-triangle` as the name for this package.
NB: Previously the linter complained for `recipes/python-triangle` just because the actual name in the recipe was `triangle`.

File: recipes/pytriangle/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,39 @@
+{% set name = "pytriangle" %}
+{% set version = "20200424" %}
+
+package:
+  name: {{ name|lower }}
```
Comment (mbargull) [inline recipes/pytriangle/meta.yaml:5]:
```suggestion
{% set version = "20200424" %}

package:
  name: python-triangle
```

File: recipes/pytriangle/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+  script: {{ PYTHON }} -m pip install . -vv
+
+requirements:
+  host:
+    - numpy
+    - pip
+    - python
+  run:
+    - numpy
+    - python
```
Comment (mbargull) [inline recipes/pytriangle/meta.yaml:23]:
```suggestion
requirements:
  build:
    - {{ compiler('c') }}
  host:
    - pip
    - python
  run:
    - numpy
    - python
```

**Comment by rbeucher** [comment]
Thanks @mbargull !

**Comment by rbeucher** [comment]
@mbargull should be good to merge

**Comment by mbargull** [comment]
I checked the licenses (which we should do, being (re-) distributors) and it looks rather disappointing :(.
The creator of the Python wrapper chose a LGPL for their code but the underlying library has a custom non-commercial license. Since -- AFAIK -- (L)GPL does not work with imposing additional restrictions (such as prohibiting commercial use), it is not possible to distribute both the wrapper as well as Triangle together.
The wrapper also vendors (i.e., includes and not just dynamically links to) Triangle, meaning any form of it (which also includes their PyPI distributions, esp. the wheels) does not have a valid license for distribution.

@rbeucher, I'm sorry, but I think you'd have to consult upstream regarding changing their license so that distribution and not just usage of their software is allowed.

**Comment by rbeucher** [comment]
OK. Don't be sorry. Let's try to sort that out.

**Pull request closed without merging**